tensorflow/python/estimator/canned/boosted_trees_test.py

```diff
@@ -983,12 +1113,52 @@ def testExperimentalFeatureImportancesWithIndicatorColumn(self):
                               'categorical_indicator:bad']
     feature_names, importances = est.experimental_feature_importances(normalize=False)
     self.assertAllEqual(feature_names_expected, feature_names)
-    self.assertAllClose([2.5, 1.5, 1.0, 0.0], importances)
+    self.assertAllClose([5.0, 3.0, 2.0, 0.0], importances)
 
     feature_names, importances = est.experimental_feature_importances(normalize=True)
     self.assertAllEqual(feature_names_expected, feature_names)
     self.assertAllClose([0.5, 0.3, 0.2, 0.0], importances)
 
+  def testNegativeFeatureImportances(self):
+    est = boosted_trees.BoostedTreesClassifier(
+        feature_columns=self._feature_columns,
+        n_batches_per_layer=1,
+        n_trees=1,
+        max_depth=5)
+
+    tree_ensemble_text = """
+        trees {
+          nodes {
+            bucketized_split {
+              feature_id: 1
+              left_id: 1
+              right_id: 2
+            }
+            metadata {
+              gain: -5.0
```

Comment:
it is mathematically impossible to get a negative gain - it is g^2/h. But fine, you can keep this around